packages/core/src/events/input/load/vectorstore/baseretriever.ts

```diff
@@ -0,0 +1,54 @@
+
+import { Context } from "../docs/context.js";
+import { Callable } from "../../../../record/callable.js";
+import { CallableConfig } from "../../../../record/callable.js";
+
+
+export interface BaseRetrieverInput {
+    callbacks?: any;
+    tags?: string[];
+    metadata?: Record<string, unknown>;
+    verbose?: boolean;
+}
+
+
+export abstract class BaseRetriever extends Callable<string, Context[]> {
+  callbacks?: any;
+
+  tags?: string[];
+
+  metadata?: Record<string, unknown>;
+
+  verbose?: boolean;
```

Comment:
I don't think you need to add these attribute if you are not modifying them? Please let me know if I'm wrong.

Reply:
However, some of the attribute that you probably want to add here. For example, `_namespace`, `_isSerializable`, `_name()`. Please check the [test case about Calllable](https://github.com/VictorS67/encre/blob/5c3e8e823bca4d3d974695e065e7fee6d599d28c/packages/core/src/record/tests/callable.test.ts#L19) to see how to customize a Callable class.

Reply:
Got it, will modify